engine/classes/ElggEntity.php

```diff
@@ -1068,6 +1068,32 @@ public function canEdit($user_guid = 0) {
 	}
 
 	/**
+	 * Can a user delete this entity?
+	 *
+	 * @tip Can be overridden by registering for the permissions_check:delete plugin hook.
+	 *
+	 * @param int $user_guid The user GUID, optionally (default: logged in user)
+	 *
+	 * @return bool Whether this entity is deletable by the given user.
+	 * @since 1.8.0
+	 * @see elgg_set_ignore_access()
+	 */
+	public function canDelete($user_guid = 0) {
+		$user_guid = (int) $user_guid;
+
+		if (!$user_guid) {
+			$user_guid = _elgg_services()->session->getLoggedInUserGuid();
+		}
+
+		$user = _elgg_services()->entityTable->get($user_guid, 'user');
+		
```

Comment:
nope



> dont want to quick return on no user (canEdit() doesn't do it) because otherwise you can not overrule it... that is what the hook is for.